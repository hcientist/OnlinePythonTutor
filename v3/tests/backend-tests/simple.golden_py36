{
  "code": "x = 5\ny = 10\nz = x * y\nprint(\"HELLO WORLD\")\nfor i in range(10):\n  print(z * i)\n\n",
  "trace": [
    {
      "line": 1,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {},
      "ordered_globals": [],
      "stack_to_render": [],
      "heap": {},
      "stdout": ""
    },
    {
      "line": 2,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "x": 5
      },
      "ordered_globals": [
        "x"
      ],
      "stack_to_render": [],
      "heap": {},
      "stdout": ""
    },
    {
      "line": 3,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "x": 5,
        "y": 10
      },
      "ordered_globals": [
        "x",
        "y"
      ],
      "stack_to_render": [],
      "heap": {},
      "stdout": ""
    },
    {
      "line": 4,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "x": 5,
        "y": 10,
        "z": 50
      },
      "ordered_globals": [
        "x",
        "y",
        "z"
      ],
      "stack_to_render": [],
      "heap": {},
      "stdout": ""
    },
    {
      "line": 5,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "x": 5,
        "y": 10,
        "z": 50
      },
      "ordered_globals": [
        "x",
        "y",
        "z"
      ],
      "stack_to_render": [],
      "heap": {},
      "stdout": "HELLO WORLD\n"
    },
    {
      "line": 6,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "x": 5,
        "y": 10,
        "z": 50,
        "i": 0
      },
      "ordered_globals": [
        "x",
        "y",
        "z",
        "i"
      ],
      "stack_to_render": [],
      "heap": {},
      "stdout": "HELLO WORLD\n"
    },
    {
      "line": 5,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "x": 5,
        "y": 10,
        "z": 50,
        "i": 0
      },
      "ordered_globals": [
        "x",
        "y",
        "z",
        "i"
      ],
      "stack_to_render": [],
      "heap": {},
      "stdout": "HELLO WORLD\n0\n"
    },
    {
      "line": 6,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "x": 5,
        "y": 10,
        "z": 50,
        "i": 1
      },
      "ordered_globals": [
        "x",
        "y",
        "z",
        "i"
      ],
      "stack_to_render": [],
      "heap": {},
      "stdout": "HELLO WORLD\n0\n"
    },
    {
      "line": 5,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "x": 5,
        "y": 10,
        "z": 50,
        "i": 1
      },
      "ordered_globals": [
        "x",
        "y",
        "z",
        "i"
      ],
      "stack_to_render": [],
      "heap": {},
      "stdout": "HELLO WORLD\n0\n50\n"
    },
    {
      "line": 6,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "x": 5,
        "y": 10,
        "z": 50,
        "i": 2
      },
      "ordered_globals": [
        "x",
        "y",
        "z",
        "i"
      ],
      "stack_to_render": [],
      "heap": {},
      "stdout": "HELLO WORLD\n0\n50\n"
    },
    {
      "line": 5,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "x": 5,
        "y": 10,
        "z": 50,
        "i": 2
      },
      "ordered_globals": [
        "x",
        "y",
        "z",
        "i"
      ],
      "stack_to_render": [],
      "heap": {},
      "stdout": "HELLO WORLD\n0\n50\n100\n"
    },
    {
      "line": 6,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "x": 5,
        "y": 10,
        "z": 50,
        "i": 3
      },
      "ordered_globals": [
        "x",
        "y",
        "z",
        "i"
      ],
      "stack_to_render": [],
      "heap": {},
      "stdout": "HELLO WORLD\n0\n50\n100\n"
    },
    {
      "line": 5,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "x": 5,
        "y": 10,
        "z": 50,
        "i": 3
      },
      "ordered_globals": [
        "x",
        "y",
        "z",
        "i"
      ],
      "stack_to_render": [],
      "heap": {},
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n"
    },
    {
      "line": 6,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "x": 5,
        "y": 10,
        "z": 50,
        "i": 4
      },
      "ordered_globals": [
        "x",
        "y",
        "z",
        "i"
      ],
      "stack_to_render": [],
      "heap": {},
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n"
    },
    {
      "line": 5,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "x": 5,
        "y": 10,
        "z": 50,
        "i": 4
      },
      "ordered_globals": [
        "x",
        "y",
        "z",
        "i"
      ],
      "stack_to_render": [],
      "heap": {},
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n200\n"
    },
    {
      "line": 6,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "x": 5,
        "y": 10,
        "z": 50,
        "i": 5
      },
      "ordered_globals": [
        "x",
        "y",
        "z",
        "i"
      ],
      "stack_to_render": [],
      "heap": {},
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n200\n"
    },
    {
      "line": 5,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "x": 5,
        "y": 10,
        "z": 50,
        "i": 5
      },
      "ordered_globals": [
        "x",
        "y",
        "z",
        "i"
      ],
      "stack_to_render": [],
      "heap": {},
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n200\n250\n"
    },
    {
      "line": 6,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "x": 5,
        "y": 10,
        "z": 50,
        "i": 6
      },
      "ordered_globals": [
        "x",
        "y",
        "z",
        "i"
      ],
      "stack_to_render": [],
      "heap": {},
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n200\n250\n"
    },
    {
      "line": 5,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "x": 5,
        "y": 10,
        "z": 50,
        "i": 6
      },
      "ordered_globals": [
        "x",
        "y",
        "z",
        "i"
      ],
      "stack_to_render": [],
      "heap": {},
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n200\n250\n300\n"
    },
    {
      "line": 6,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "x": 5,
        "y": 10,
        "z": 50,
        "i": 7
      },
      "ordered_globals": [
        "x",
        "y",
        "z",
        "i"
      ],
      "stack_to_render": [],
      "heap": {},
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n200\n250\n300\n"
    },
    {
      "line": 5,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "x": 5,
        "y": 10,
        "z": 50,
        "i": 7
      },
      "ordered_globals": [
        "x",
        "y",
        "z",
        "i"
      ],
      "stack_to_render": [],
      "heap": {},
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n200\n250\n300\n350\n"
    },
    {
      "line": 6,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "x": 5,
        "y": 10,
        "z": 50,
        "i": 8
      },
      "ordered_globals": [
        "x",
        "y",
        "z",
        "i"
      ],
      "stack_to_render": [],
      "heap": {},
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n200\n250\n300\n350\n"
    },
    {
      "line": 5,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "x": 5,
        "y": 10,
        "z": 50,
        "i": 8
      },
      "ordered_globals": [
        "x",
        "y",
        "z",
        "i"
      ],
      "stack_to_render": [],
      "heap": {},
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n200\n250\n300\n350\n400\n"
    },
    {
      "line": 6,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "x": 5,
        "y": 10,
        "z": 50,
        "i": 9
      },
      "ordered_globals": [
        "x",
        "y",
        "z",
        "i"
      ],
      "stack_to_render": [],
      "heap": {},
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n200\n250\n300\n350\n400\n"
    },
    {
      "line": 5,
      "event": "step_line",
      "func_name": "<module>",
      "globals": {
        "x": 5,
        "y": 10,
        "z": 50,
        "i": 9
      },
      "ordered_globals": [
        "x",
        "y",
        "z",
        "i"
      ],
      "stack_to_render": [],
      "heap": {},
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n200\n250\n300\n350\n400\n450\n"
    },
    {
      "line": 5,
      "event": "return",
      "func_name": "<module>",
      "globals": {
        "x": 5,
        "y": 10,
        "z": 50,
        "i": 9
      },
      "ordered_globals": [
        "x",
        "y",
        "z",
        "i"
      ],
      "stack_to_render": [],
      "heap": {},
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n200\n250\n300\n350\n400\n450\n"
    }
  ]
}
