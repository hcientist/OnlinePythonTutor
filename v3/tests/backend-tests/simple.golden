{
  "code": "x = 5\ny = 10\nz = x * y\nprint(\"HELLO WORLD\")\nfor i in range(10):\n  print(z * i)\n\n", 
  "trace": [
    {
      "ordered_globals": [], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {}, 
      "heap": {}, 
      "line": 1, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "x": 5
      }, 
      "heap": {}, 
      "line": 2, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "y": 10, 
        "x": 5
      }, 
      "heap": {}, 
      "line": 3, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z"
      ], 
      "stdout": "", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "y": 10, 
        "x": 5, 
        "z": 50
      }, 
      "heap": {}, 
      "line": 4, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z"
      ], 
      "stdout": "HELLO WORLD\n", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "y": 10, 
        "x": 5, 
        "z": 50
      }, 
      "heap": {}, 
      "line": 5, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "i"
      ], 
      "stdout": "HELLO WORLD\n", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "i": 0, 
        "x": 5, 
        "z": 50, 
        "y": 10
      }, 
      "heap": {}, 
      "line": 6, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "i"
      ], 
      "stdout": "HELLO WORLD\n0\n", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "i": 0, 
        "x": 5, 
        "z": 50, 
        "y": 10
      }, 
      "heap": {}, 
      "line": 5, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "i"
      ], 
      "stdout": "HELLO WORLD\n0\n", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "i": 1, 
        "x": 5, 
        "z": 50, 
        "y": 10
      }, 
      "heap": {}, 
      "line": 6, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "i"
      ], 
      "stdout": "HELLO WORLD\n0\n50\n", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "i": 1, 
        "x": 5, 
        "z": 50, 
        "y": 10
      }, 
      "heap": {}, 
      "line": 5, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "i"
      ], 
      "stdout": "HELLO WORLD\n0\n50\n", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "i": 2, 
        "x": 5, 
        "z": 50, 
        "y": 10
      }, 
      "heap": {}, 
      "line": 6, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "i"
      ], 
      "stdout": "HELLO WORLD\n0\n50\n100\n", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "i": 2, 
        "x": 5, 
        "z": 50, 
        "y": 10
      }, 
      "heap": {}, 
      "line": 5, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "i"
      ], 
      "stdout": "HELLO WORLD\n0\n50\n100\n", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "i": 3, 
        "x": 5, 
        "z": 50, 
        "y": 10
      }, 
      "heap": {}, 
      "line": 6, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "i"
      ], 
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "i": 3, 
        "x": 5, 
        "z": 50, 
        "y": 10
      }, 
      "heap": {}, 
      "line": 5, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "i"
      ], 
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "i": 4, 
        "x": 5, 
        "z": 50, 
        "y": 10
      }, 
      "heap": {}, 
      "line": 6, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "i"
      ], 
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n200\n", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "i": 4, 
        "x": 5, 
        "z": 50, 
        "y": 10
      }, 
      "heap": {}, 
      "line": 5, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "i"
      ], 
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n200\n", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "i": 5, 
        "x": 5, 
        "z": 50, 
        "y": 10
      }, 
      "heap": {}, 
      "line": 6, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "i"
      ], 
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n200\n250\n", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "i": 5, 
        "x": 5, 
        "z": 50, 
        "y": 10
      }, 
      "heap": {}, 
      "line": 5, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "i"
      ], 
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n200\n250\n", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "i": 6, 
        "x": 5, 
        "z": 50, 
        "y": 10
      }, 
      "heap": {}, 
      "line": 6, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "i"
      ], 
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n200\n250\n300\n", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "i": 6, 
        "x": 5, 
        "z": 50, 
        "y": 10
      }, 
      "heap": {}, 
      "line": 5, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "i"
      ], 
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n200\n250\n300\n", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "i": 7, 
        "x": 5, 
        "z": 50, 
        "y": 10
      }, 
      "heap": {}, 
      "line": 6, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "i"
      ], 
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n200\n250\n300\n350\n", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "i": 7, 
        "x": 5, 
        "z": 50, 
        "y": 10
      }, 
      "heap": {}, 
      "line": 5, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "i"
      ], 
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n200\n250\n300\n350\n", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "i": 8, 
        "x": 5, 
        "z": 50, 
        "y": 10
      }, 
      "heap": {}, 
      "line": 6, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "i"
      ], 
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n200\n250\n300\n350\n400\n", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "i": 8, 
        "x": 5, 
        "z": 50, 
        "y": 10
      }, 
      "heap": {}, 
      "line": 5, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "i"
      ], 
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n200\n250\n300\n350\n400\n", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "i": 9, 
        "x": 5, 
        "z": 50, 
        "y": 10
      }, 
      "heap": {}, 
      "line": 6, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "i"
      ], 
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n200\n250\n300\n350\n400\n450\n", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "i": 9, 
        "x": 5, 
        "z": 50, 
        "y": 10
      }, 
      "heap": {}, 
      "line": 5, 
      "event": "step_line"
    }, 
    {
      "ordered_globals": [
        "x", 
        "y", 
        "z", 
        "i"
      ], 
      "stdout": "HELLO WORLD\n0\n50\n100\n150\n200\n250\n300\n350\n400\n450\n", 
      "func_name": "<module>", 
      "stack_to_render": [], 
      "globals": {
        "i": 9, 
        "y": 10, 
        "z": 50, 
        "x": 5
      }, 
      "heap": {}, 
      "line": 5, 
      "event": "return"
    }
  ]
}
